### PR TITLE
Adding system hardening guide

### PR DESCRIPTION
**Pull request by hammadahmedirdeto**
Adding more content to the Hardening section of the security blueprint.

### CODE REVIEW

**Review by dl9pf — CHANGES_REQUESTED** [review]
TLDR:
As we have multiple options that are different for DEBUG/RELEASE, we might extend the document with a longer comparison table with all options in the end.

Also we need a full review here to not contradict subsystem like the application framework or sota/ostree.




Here is my feedback in detail:

- Disable namespace support
See: https://github.com/automotive-grade-linux/docs-agl/blob/2bf24424d624d478293c7badd7761328745df17e/docs/sec-blueprint/08-Hardening.md#disable-namespace-support

JSMoeller: IIRC Systemd and other tools require this. This needs to be checked before we recommend it!

- Disable /proc/config.gz
See: https://github.com/automotive-grade-linux/docs-agl/blob/2bf24424d624d478293c7badd7761328745df17e/docs/sec-blueprint/08-Hardening.md#disable-procconfiggz

JSMoeller: To be honest, I don't like the way this expressed. An attacker will likely know the CPU/BasicHW so not having the /proc/config.gz is a minor obstacle that costs just time IMHO. And moreover he needs to be 'on' the target already to catch that. Better differ between DEBUG and PRODUCTION build here, so push that down to the section/table with DEBUG/RELEASE. 


- Disable Sysctl syscall support
"Enabling this will result in code being included that is hard to maintain and not well tested"

JSMoeller: A little more explanation as to why - this is very brief.

- Console & Remote Access
See: https://github.com/automotive-grade-linux/docs-agl/blob/2bf24424d624d478293c7badd7761328745df17e/docs/sec-blueprint/08-Hardening.md#console--remote-access-1

JSMoeller: We are in a systemd environment, there is no inittab like this. 

- Partition Mount Options
See: https://github.com/automotive-grade-linux/docs-agl/blob/2bf24424d624d478293c7badd7761328745df17e/docs/sec-blueprint/08-Hardening.md#partition-mount-options

JSMoeller: Mount options need to be aligned with ostree recommendations/needs.

- Disable checkpoint/restore
See: https://github.com/automotive-grade-linux/docs-agl/blob/2bf24424d624d478293c7badd7761328745df17e/docs/sec-blueprint/08-Hardening.md#disable-checkpointrestore

JSMoeller: This needs to be reviewed in conjunction with the application lifecycle. If I'm not mistaken checkpoint/restore was a required feature.


- Removal or Non-Inclusion of Utilities
See: https://github.com/automotive-grade-linux/docs-agl/blob/2bf24424d624d478293c7badd7761328745df17e/docs/sec-blueprint/08-Hardening.md#removal-or-non-inclusion-of-utilities

JSMoeller: we need to double-check this list (ssh vs dropbear, init vs. systemd)

- Enable Iptables (firewall)

JSMoeller: we're using systemd, so iptables or nftables or firewalld ... we should recommend the long-term supported technique that fits into the env best.

**Comment by dl9pf** [comment]
We need ATS in the reviewer list ...

**Comment by fulup-bzh** [comment]
For name space or systemd it is obvious that we need them. Firewall iptables/nftables should be generated automatically by the application framework at application installation time. Removing checkpoint/restore might not be possible to enable application pause&criu.

The list of command to include/remove seems outdated "netstat" as been replace by "ip". On system using systemd /etc/init/rcS.d /etc/init.d/inittab is empty

For the other point I agree with JSMoeller

**Comment by hammadahmedirdeto** [comment]
Updated based on review comments provided.
As far as DEBUG/RELEASE kernel configuration goes, I think we should leave this document as hardening guide for production kernels and shouldn't worry about what configs a dev turns on to debug a problem.

**Review by dl9pf — CHANGES_REQUESTED** [review]
Hi, my 0,02:
- We need ATS to review this from a OTA perspective. Some of the changes might still collide there.

- Here is my feedback in detail:

1. Disable namespace support  -  **Done, was changed**
2. Disable /proc/config.gz
    See: https://github.com/hammadahmedirdeto/docs-agl/blob/0996761cc552e698b97f2e07f6894a46ab82345c/docs/sec-blueprint/08-Hardening.md#disable-procconfiggz
JSMoeller: To be honest, I don't like the way this expressed. An attacker will likely know the CPU/BasicHW so not having the /proc/config.gz is a minor obstacle that costs just time IMHO. And moreover he needs to be 'on' the target already to catch that. Better differ between DEBUG and PRODUCTION build here, so push that down to the section/table with DEBUG/RELEASE.
**JSMoeller: Still an issue**

3. Disable Sysctl syscall support
    "Enabling this will result in code being included that is hard to maintain and not well tested"
**JSMoeller: A little more explanation as to why - this is very brief.**

4. Console & Remote Access - **Done, was changed**

5. Partition Mount Options 
    See: https://github.com/hammadahmedirdeto/docs-agl/blob/0996761cc552e698b97f2e07f6894a46ab82345c/docs/sec-blueprint/08-Hardening.md#partition-mount-options
JSMoeller: **Mount options need to be aligned with ostree recommendations/needs.**
JSMoeller: **Add ATS to reviewers for this.**

6. Disable checkpoint/restore - **Done, was changed**

7. Removal or Non-Inclusion of Utilities
    See: https://github.com/hammadahmedirdeto/docs-agl/blob/0996761cc552e698b97f2e07f6894a46ab82345c/docs/sec-blueprint/08-Hardening.md#removal-or-non-inclusion-of-utilities
JSMoeller: we need to double-check this list (ssh vs dropbear, init vs. systemd, klog vs journalctl, stbhotplug, 

8. Enable Iptables (firewall) - **section removed, still we should recommend something**

9. disable bpf jit
  See: https://github.com/hammadahmedirdeto/docs-agl/blob/0996761cc552e698b97f2e07f6894a46ab82345c/docs/sec-blueprint/08-Hardening.md#disable-bpf-jit
JSMoeller: Unclear statement, we're not using mips. This might actually help in firewalling, so depends on choice of firewall here.

10. MIPS arch mentioned
 See: https://github.com/hammadahmedirdeto/docs-agl/blob/0996761cc552e698b97f2e07f6894a46ab82345c/docs/sec-blueprint/08-Hardening.md#kernel-hardening-1
JSMoeller: We're not using MIPS, just  x86 and arm (32/64bit) right now.


11.  –static 
 `Dynamic linking shall not be allowed. This will avoid user from replacing a library with malicious library. All libraries should be linked statically.`
 See: https://github.com/hammadahmedirdeto/docs-agl/blob/0996761cc552e698b97f2e07f6894a46ab82345c/docs/sec-blueprint/08-Hardening.md#library-linking
JSMoeller: This is **not going to fly** in the way things are done right now. Also if we are on a filesystem managed by ostree, a user cannot change a file in /lib or /usr/lib as it is under control of ostree.

File: docs/sec-blueprint/08-Hardening.md [hunk 1]
```diff
@@ -41,16 +41,193 @@ invalid binaries on the device, and to prevent copying of security
...
+and configuration of the root filesystem.
+
+**Application Hardening:** Best practices to apply to the build and
+release of user space applications, in order to reduce the number of
+attack surfaces used by potential attackers.
+
+Secure Boot Software Flow Steps
+-------------------------------
+
+1.  After power on, the secure processor will perform the verification
```
Comment (jobol) [inline docs/sec-blueprint/08-Hardening.md]: The use of a 'secure processor' or of some native mechanism to achieve this operation is a strong requirement that should be explained first.

File: docs/sec-blueprint/08-Hardening.md [hunk 2]
```diff
@@ -41,16 +41,193 @@ invalid binaries on the device, and to prevent copying of security
...
+    of the Stage 1 boot image, the stage 2 boot image and the Secure
+    loader image.
+
+    a.  If any of the images fail the verification process the device
+        will restart.
```
Comment (jobol) [inline docs/sec-blueprint/08-Hardening.md]: I'm not sure that restarting the device is a solution! IMHO it should stop.

File: docs/sec-blueprint/08-Hardening.md [hunk 3]
```diff
@@ -41,16 +41,193 @@ invalid binaries on the device, and to prevent copying of security
...
+3.  The Stage 1 boot process will perform processor initialization, and
+    then initiate the Stage 2 boot process.
+
+4.  The Stage 2 boot process will initiate the Secure Loader, which will
+    process any customer specific customizations (e.g. front panel
+    of ECU, USB based image updates, etc).
+
+5.  The Secure Loader will check to determine if there are any updates
+    to be processed. If the update settings indicate that an upgrade
+    should occur then the Secure Loader will will determine the correct
```
Comment (jobol) [inline docs/sec-blueprint/08-Hardening.md:80]: typo: will will -> will

File: docs/sec-blueprint/08-Hardening.md [hunk 4]
```diff
@@ -41,16 +41,193 @@ invalid binaries on the device, and to prevent copying of security
...
+    action based on the nature of the upgrades:
+
+    a.  If the Secure Loader determines that an upgrade was performed
+        (or attempted), it will initiate the reboot process.
```
Comment (jobol) [inline docs/sec-blueprint/08-Hardening.md:84]: There is probably something missing here because if the state is not changed, the reboot will reach the same conditions and an infinite loop will  aoccur (at least in my smoggy brain)

Anyway, advise of ATS is really needed here. I think that there is no conflict with their approach but I would get the feedback.

File: docs/sec-blueprint/08-Hardening.md [hunk 5]
```diff
@@ -41,16 +41,193 @@ invalid binaries on the device, and to prevent copying of security
...
+
+    b.  If no upgrades were processed: then the Secure Loader will pass
+        control back to the Stage 2 boot process for further processing
+
+6.  The Stage 2 boot process will continue with the boot process, by
+    performing a verification of the kernel image prior to the load of
+    that image
+
+    a.  If the kernel image verification fails, the Stage 2 boot loader
+        will initiate a reboot
```
Comment (jobol) [inline docs/sec-blueprint/08-Hardening.md]: or a stop

File: docs/sec-blueprint/08-Hardening.md [hunk 6]
```diff
@@ -41,16 +41,193 @@ invalid binaries on the device, and to prevent copying of security
...
+
+    a.  If the Secure Loader determines that an upgrade was performed
+        (or attempted), it will initiate the reboot process.
+
+    b.  If no upgrades were processed: then the Secure Loader will pass
+        control back to the Stage 2 boot process for further processing
+
+6.  The Stage 2 boot process will continue with the boot process, by
+    performing a verification of the kernel image prior to the load of
+    that image
```
Comment (jobol) [inline docs/sec-blueprint/08-Hardening.md:91]: What is the exact meaning of "kernel image"?
Is it "the kernel" or is it "the kernel and the root kernel data/partition"?
Verifying kernel is probably not enough because further stages inside the kernel will surely need to check and validate data. To achieve it there are 2 possibililties at least: using a security processor that can validate the data or rely on a trusted data environment. The later means that the data must be validated in the same time.

File: docs/sec-blueprint/08-Hardening.md [hunk 7]
```diff
@@ -41,16 +41,193 @@ invalid binaries on the device, and to prevent copying of security
...
+    a.  If the kernel image verification fails, the Stage 2 boot loader
+        will initiate a reboot
+
+8.  The Stage 2 boot loader will load the successfully verified kernel
+    image and boot the linux OS
+
+9.  The booted Linux OS will perform the normal Linux init sequence
+
+10. The Linux init process will start the required applications and
+    services as described in the init process and present on the rootfs.
```
Comment (jobol) [inline docs/sec-blueprint/08-Hardening.md:103]: Integrity of the booted files is not clearly adressed.

File: docs/sec-blueprint/08-Hardening.md [hunk 8]
```diff
@@ -41,16 +41,193 @@ invalid binaries on the device, and to prevent copying of security
...
 
 Requirements
 ============
 
   For the purposes of reference and explanation, we are providing guidance
   on how to configure an embedded device that runs with a linux 3. 10.17
-  Linux kernel.  
+  Linux kernel, and includes the use of U-Boot as the *Stage 2*
```
Comment (jobol) [inline docs/sec-blueprint/08-Hardening.md:109]: Is U-Boot ARM specific ? Nevertheless I don't think that U-Boot is mandatory.

File: docs/sec-blueprint/08-Hardening.md [hunk 9]
```diff
@@ -41,16 +41,193 @@ invalid binaries on the device, and to prevent copying of security
...
+Hardened Boot
+-------------
+
+### Boot image selection
+
+The boot process shall be uninterruptable and shall irrevocably boot the
+image as specified in the boot environment.
+
+In U-Boot set the “**bootdelay**” environment variable and/or define
+CONFIG\_BOOTDELAY to -2.
```
Comment (jobol) [inline docs/sec-blueprint/08-Hardening.md:122]: Reference needed and below. I quickly dug http://www.denx.de/wiki/U-Boot/WebHome but I found nothing.

File: docs/sec-blueprint/08-Hardening.md [hunk 10]
```diff
@@ -41,16 +41,193 @@ invalid binaries on the device, and to prevent copying of security
...
+
+```
+CONFIG_FIT: enables support for Flat Image Tree (FIT) uImage format.
+CONFIG_FIT_SIGNATURE: enables signature verification of FIT images.
+CONFIG_RSA: enables RSA algorithm used for FIT image verifitcation.
+CONFIG_OF_CONTROL: enables Flattened Device Tree (FDT) configuration.
+CONFIG_OF_SEPARATE: enables separate build of u-Boot from the device tree.
+CONFIG_DEFAULT_DEVICE_TREE: specifies the default Device Tree used for the
+run-time configuration of U-Boot.
+```
```
Comment (jobol) [inline docs/sec-blueprint/08-Hardening.md:141]: It partly answers some questions I had before: this is the flat image tree that is checked.

File: docs/sec-blueprint/08-Hardening.md [hunk 11]
```diff
@@ -41,16 +41,193 @@ invalid binaries on the device, and to prevent copying of security
...
+### Field upgrades
+
+Field upgrades can be achieved securely by using a Secure Loader.
+This loader will authenticate an incoming image (USB,Serial, Network)
+prior to writing it to the flash memory on the device. It should not be
+possible to write to flash from bootloader (U-Boot). Note that because
+ USB support is to be disabled within the sboot/U-Boot code, the board
+specific implementation of the Secure Loader will have to manage the
+entire USB initialization, enumeration, and read/write access to the
+mass storage device.
```
Comment (jobol) [inline docs/sec-blueprint/08-Hardening.md:190]: Adding the possibility to achieve the operation from storage is not presented here.

File: docs/sec-blueprint/08-Hardening.md [hunk 12]
```diff
@@ -41,16 +41,193 @@ invalid binaries on the device, and to prevent copying of security
...
+#define CONFIG_ENV_IS_IN_ONENAND
+#define CONFIG_ENV_IS_IN_SPI_FLASH
+#define CONFIG_ENV_IS_IN_REMOTE
+#define CONFIG_ENV_IS_IN_UBI
+```
+
+and include the following definition:
+
+```
+#define** CONFIG_ENV_IS_NOWHERE
```
Comment (jobol) [inline docs/sec-blueprint/08-Hardening.md:228]: Can you precise a location where this define is to be made

File: docs/sec-blueprint/08-Hardening.md [hunk 13]
```diff
@@ -365,6 +531,204 @@ Kernel Hardening
...
   CONFIG_MODULE_FORCE_LOAD=n
 ```
+### System Services
+
+#### Console & Remote Access
+
+-   The kernel console interfaces shall be disabled. Do not pass any statements
+    of the following kind (e.g. console=ttyS0 console=tty0) on the kernel
+    command line. All of the console=&lt;interface&gt; statements should be
+    stripped and removed from the kernel command line.
```
Comment (jobol) [inline docs/sec-blueprint/08-Hardening.md:541]: There is something to say about the command line: because console is prohibited, no one should be able to enter any command line (that come from uboot hardening). Nevertheless, kernel should also be hardened to disable command line arguments.

File: docs/sec-blueprint/08-Hardening.md [hunk 14]
```diff
@@ -365,6 +531,204 @@ Kernel Hardening
...
+    be disabled.
+
+-   In the /etc/ssh/sshd\_config file:
+
+    **PermitRootLogin no**
+
+    Or if running dropbear ssh server, in the /etc/config/dropbear config
+    file:
+ 
+    **RootLogin no**
```
Comment (jobol) [inline docs/sec-blueprint/08-Hardening.md]: SSH and or drop bear or equivalent must be removed.

File: docs/sec-blueprint/08-Hardening.md [hunk 15]
```diff
@@ -365,6 +531,204 @@ Kernel Hardening
...
+### Debuggers shall not be present on the file system
+
+  No debuggers shall be present on the file system. This includes, but
+  is not limited to, the GNU Debugger client/server (commonly known in
+  their short form names such as the gdb and gdbserver executable
+  binaries respectively), or the LLDB next generation debugger.
+  Including these binaries as part of the file system will facilitate an
+  attacker’s ability to reverse engineer and debug (either locally or
+  remotely) any process that is currently executing on the device.
+
```
Comment (jobol) [inline docs/sec-blueprint/08-Hardening.md:689]: this include tcf-agent that is currently shipped with devel version

File: docs/sec-blueprint/08-Hardening.md [hunk 16]
```diff
@@ -580,24 +1014,228 @@ Kernel Hardening
...
+incur some performance loss during initial load while symbols are
+resolved, but this shouldn't be an issue for daemons.
+
+### Library linking
+
+  **–static**
+
+Dynamic linking shall not be allowed. This will avoid user from
+replacing a library with malicious library. All libraries should be
+linked statically.
```
Comment (jobol) [inline docs/sec-blueprint/08-Hardening.md]: Ideally yes but in practice this will probably not happen. Even more, because AGL allows dynamic loading, it can create bugs.
I propose to keep the idea but to rewrite as a recommandation. 
As a consequence it would be benefical for static linking to enforce the options -ffunction-sections -fdata-sections with the goal of reducing the overall size.

File: docs/sec-blueprint/08-Hardening.md [hunk 17]
```diff
@@ -580,24 +1014,228 @@ Kernel Hardening
...
+respective environment.
+
+|  **Utility Name** |  **Location**                             |  **Debug Environment** |   **Production Environment** |
+|-------------------|-------------------------------------------|------------------------|------------------------------|
+|  Strace           |  /bin/trace                               |  INCLUDE               |  EXCLUDE                     |
+|  Klogd            |  /sbin/klogd                              |  INCLUDE               |  EXCLUDE                     |
+|  Syslogd(logger)  |  /bin/logger                              |  INCLUDE               |  EXCLUDE                     |
+|  Gdbserver        |  /bin/gdbserver                           |  INCLUDE               |  EXCLUDE                     |
+|  Dropbear         |  Remove “dropbear” from ‘/etc/init.d/rcs’ |  EXCLUDE               |  EXCLUDE                     |
+|  SSH              |  NA                                       |  EXCLUDE               |  EXCLUDE                     |
```
Comment (jobol) [inline docs/sec-blueprint/08-Hardening.md]: no SSH in debug! I would prefer to have one.

File: docs/sec-blueprint/08-Hardening.md [hunk 18]
```diff
@@ -580,24 +1014,228 @@ Kernel Hardening
...
+|  Editors (vi)     |  /bin/vi                                  |  INCLUDE               |  EXCLUDE                     |
+|  Dmesg            |  /bin/dmesg                               |  INCLUDE               |  EXCLUDE                     |
+|  UART             |  /proc/tty/driver/                        |  INCLUDE               |  EXCLUDE                     |
+|  Hexdump          |  /bin/hexdump                             |  INCLUDE               |  EXCLUDE                     |
+|  Dnsdomainname    |  /bin/dnsdomainname                       |  EXCLUDE               |  EXCLUDE                     |
+|  Hostname         |  /bin/hostname                            |  INCLUDE               |  EXCLUDE                     |
+|  Pmap             |  /bin/pmap                                |  INCLUDE               |  EXCLUDE                     |
+|  su               |  /bin/su                                  |  EXCLUDE               |  EXCLUDE                     |
```
Comment (jobol) [inline docs/sec-blueprint/08-Hardening.md]: we need su in debug because AGL feature users

File: docs/sec-blueprint/08-Hardening.md [hunk 19]
```diff
@@ -580,24 +1014,228 @@ Kernel Hardening
...
+utilities are not required by the device then those should be removed.
+
+  **sed, awk, cut, df, dmesg, echo, fdisk, grep, mkdir, mount (vfat),
+  printf, tail, tee, test (directory), test (file)**
+
+Root Access 
+------------
+
+The main applications, those that provide the principal functionality of
+the embedded device, **should not execute** with root privileges.
```
Comment (jobol) [inline docs/sec-blueprint/08-Hardening.md]: Can you replace "root privileges" by "root identity or any capability"

File: docs/sec-blueprint/08-Hardening.md [hunk 20]
```diff
@@ -580,24 +1014,228 @@ Kernel Hardening
...
+
+If the main application are allowed to execute at elevated privileges,
```
Comment (jobol) [inline docs/sec-blueprint/08-Hardening.md]: you may replace "elevated privileges" by "any capability"

File: docs/sec-blueprint/08-Hardening.md [hunk 21]
```diff
@@ -580,24 +1014,228 @@ Kernel Hardening
...
+then the entire system is at the mercy of the said application’s good
+behaviour. Problems arise when an application is compromised and able to
+execute commands which could consistently and persistently compromise
+the system by implanting rogue applications.
+
+It is suggested that the middleware and the UI should run in a context
+on a user with lowered privileges and all persistent resources should be
```
Comment (jobol) [inline docs/sec-blueprint/08-Hardening.md]: no capability

File: docs/sec-blueprint/08-Hardening.md [hunk 22]
```diff
@@ -580,24 +1014,228 @@ Kernel Hardening
...
+maintained with such privileges.
+
```
Comment (jobol) [inline docs/sec-blueprint/08-Hardening.md]: by "... with such privileges" should I understand " ... without any capability" ?

File: docs/sec-blueprint/08-Hardening.md [hunk 23]
```diff
@@ -580,24 +1014,228 @@ Kernel Hardening
...
+### Ignore all broadcast message
+
+  All the IP packets that come on the address “255.255.255.255” shall be
+  ignored. This can be done through the iptables rules.
+
+### Disable IPV6 
+
+  If there are no plans of using IPV6, it is a good practice to disable
+  this support as it will reduce the size of the kernel TCP/IP stack.
```
Comment (jobol) [inline docs/sec-blueprint/08-Hardening.md:1207]: humm... isn't it time to switch to ipv6 and disable ipv4? (joke) I fully agree with the assertion.

File: docs/sec-blueprint/08-Hardening.md (hunk 1 above)
Comment (hammadahmedirdeto) [reply docs/sec-blueprint/08-Hardening.md]: Fixed

File: docs/sec-blueprint/08-Hardening.md (hunk 2 above)
Comment (hammadahmedirdeto) [reply docs/sec-blueprint/08-Hardening.md]: Fixed

File: docs/sec-blueprint/08-Hardening.md (hunk 3 above)
Comment (hammadahmedirdeto) [reply docs/sec-blueprint/08-Hardening.md:80]: Fixed

File: docs/sec-blueprint/08-Hardening.md (hunk 4 above)
Comment (hammadahmedirdeto) [reply docs/sec-blueprint/08-Hardening.md:84]: Been waiting for ATS review since December

File: docs/sec-blueprint/08-Hardening.md (hunk 6 above)
Comment (hammadahmedirdeto) [reply docs/sec-blueprint/08-Hardening.md:91]: Kernel image would means the kernel and the root kernel data/partition

File: docs/sec-blueprint/08-Hardening.md (hunk 5 above)
Comment (hammadahmedirdeto) [reply docs/sec-blueprint/08-Hardening.md]: Fixed

File: docs/sec-blueprint/08-Hardening.md (hunk 7 above)
Comment (hammadahmedirdeto) [reply docs/sec-blueprint/08-Hardening.md:103]: I am not sure what you mean here

File: docs/sec-blueprint/08-Hardening.md (hunk 8 above)
Comment (hammadahmedirdeto) [reply docs/sec-blueprint/08-Hardening.md:109]: You are right. We have only looked at U-Boot for the sake of this document, mostly because of the verified boot capability.

File: docs/sec-blueprint/08-Hardening.md (hunk 11 above)
Comment (hammadahmedirdeto) [reply docs/sec-blueprint/08-Hardening.md:190]: Can you elaborate a bit more here?

File: docs/sec-blueprint/08-Hardening.md (hunk 12 above)
Comment (hammadahmedirdeto) [reply docs/sec-blueprint/08-Hardening.md:228]: Planning to add u-boot config reference as a separate pull req. Will open an issue for this.

File: docs/sec-blueprint/08-Hardening.md (hunk 14 above)
Comment (hammadahmedirdeto) [reply docs/sec-blueprint/08-Hardening.md]: Fixed

File: docs/sec-blueprint/08-Hardening.md (hunk 16 above)
Comment (hammadahmedirdeto) [reply docs/sec-blueprint/08-Hardening.md]: Fixed

File: docs/sec-blueprint/08-Hardening.md (hunk 17 above)
Comment (hammadahmedirdeto) [reply docs/sec-blueprint/08-Hardening.md]: Fixed

File: docs/sec-blueprint/08-Hardening.md (hunk 18 above)
Comment (hammadahmedirdeto) [reply docs/sec-blueprint/08-Hardening.md]: Fixed

File: docs/sec-blueprint/08-Hardening.md (hunk 19 above)
Comment (hammadahmedirdeto) [reply docs/sec-blueprint/08-Hardening.md]: Fixed

File: docs/sec-blueprint/08-Hardening.md (hunk 21 above)
Comment (hammadahmedirdeto) [reply docs/sec-blueprint/08-Hardening.md]: Fixed

File: docs/sec-blueprint/08-Hardening.md (hunk 20 above)
Comment (hammadahmedirdeto) [reply docs/sec-blueprint/08-Hardening.md]: Fixed

File: docs/sec-blueprint/08-Hardening.md (hunk 22 above)
Comment (hammadahmedirdeto) [reply docs/sec-blueprint/08-Hardening.md]: Fixed

File: docs/sec-blueprint/08-Hardening.md (hunk 7 above)
Comment (jobol) [reply docs/sec-blueprint/08-Hardening.md:103]: You wrote "Kernel image would means the kernel and the root kernel data/partition" so it not clearly include the the extra file system data. If I refer to FHS, it means that you verify /lib but not /usr/lib, /bin but not /usr/bin.

How to call it clearly? The platform filesystem? The user partition?

My understanding is that there is by need a separation between the root/kernel/system in  one hand and the effective installed applications running on top of that. The need comes from the fact that the system will be shared by many different vehicle but the platform is specific to models+options. The later should also be checked.

Obviously it can be made differently than the gore/kernel/root/system because the kernel is trusted and it can be used to check that data. But (1) storing signature is to be addressed somewhere (2) checking it should be specified somewhere in the blue print, why not here?

File: docs/sec-blueprint/08-Hardening.md (hunk 11 above)
Comment (jobol) [reply docs/sec-blueprint/08-Hardening.md:190]: I was meaning that "This loader will authenticate an incoming image (USB,Serial, Network)" is closing the possibilities, maybe with right. I ask myself whether writing  "(USB,Serial, Network, Disk, Memory)" is possible or not or even merely "(USB,Serial, Network, eMMC, ...)"

**Review by jobol — APPROVED** [review]
